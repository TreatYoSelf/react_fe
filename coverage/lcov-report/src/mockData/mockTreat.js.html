
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/mockData/mockTreat.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">src/mockData</a> mockTreat.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const mockTreat = {
    id: 1,
    icon: 'https://www.pikpng.com/pngl/b/65-658548_source-munns-com-au-report-sun-icon-png.png',
    title: 'Frolic',
    rating: 3,
}
&nbsp;
export const mockTreatSched = {
    id: 1,
    icon: 'https://www.pikpng.com/pngl/b/65-658548_source-munns-com-au-report-sun-icon-png.png',
    title: 'Competitive Plumbing',
    date: '',
    duration: ''
}
&nbsp;
export const mockCategories = {
    getCategories: [
        {
            "name": "Outdoor",
        },
        {
            "name": "Beauty",
        },
        {
            "name": "Exercise",
        },
        {
            "name": "Home",
        },
        {
            "name": "Organizing",
        },
        {
            "name": "Food",
        },
        {
            "name": "Mindfulness",
        },
        {
            "name": "Group",
        },
        {
            "name": "Event",
        },
        {
            "name": "Enrichment",
        },
    ]
}
&nbsp;
export const mockEvents = [{
        "eventName": "Hike",
            "eventStartTime": "1394648309130.185",
            "eventEndTime": "1394651309130.185"
    },
    {
        "eventName": "Spa Day",
            "eventStartTime": "1396678309130.185",
            "eventEndTime": "1396682309130.185"
    },
    {
        "eventName": "Yoga",
            "eventStartTime": "1396678509130.185",
            "eventEndTime": "1396682309130.185"
    }];
&nbsp;
export const mockUser = {
    accessToken: "ya29.a0Ae4lvC1pqc5Cba23lW-FzAtzOJyQ9SQCWmy0ieLjCWn-8p4XZ65Ba8SLRmPIhDEGyv38kfvkT_o3uevsJ2HRUlKTvEbyibGDziIIGhUCa31zDVUUlc23o0tdp3F6c5erDMKSCDUAzzczq9N87_hZDjI6adiSMl_TQoM",
    idToken: "eyJhbGciOiJSUzI1NiIsImtpZCI6IjZmY2Y0MTMyMjQ3NjUxNTZiNDg3NjhhNDJmYWMwNjQ5NmEzMGZmNWEiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJhenAiOiI4NDI0ODY4MjkyNDItbXEwZTZkNWxwc3FsN2w5dTQxbzRjY2xhcXBycHNianAuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJhdWQiOiI4NDI0ODY4MjkyNDItbXEwZTZkNWxwc3FsN2w5dTQxbzRjY2xhcXBycHNianAuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJzdWIiOiIxMTgxMjM5MDQ3MDMzNTY3NDA1NTQiLCJlbWFpbCI6ImxhY2V5ZmFtaWx5MjAzQGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhdF9oYXNoIjoiNGdpMzVuZl9aazdhR3VLTHJSYnRhQSIsIm5hbWUiOiJMYWNleSBGYW1pbHkiLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDQuZ29vZ2xldXNlcmNvbnRlbnQuY29tLy16QkNva183Qk5wMC9BQUFBQUFBQUFBSS9BQUFBQUFBQUFBQS9BQUtXSkpQMVF3eEs3V0x4MGhRLXZZY0lId1k5dV9wbVJRL3M5Ni1jL3Bob3RvLmpwZyIsImdpdmVuX25hbWUiOiJMYWNleSIsImZhbWlseV9uYW1lIjoiRmFtaWx5IiwibG9jYWxlIjoiZW4iLCJpYXQiOjE1ODY5Njc3MTAsImV4cCI6MTU4Njk3MTMxMH0.GLKpXoTmPPIs_oFh3mmJNNY9XmosgJEqwmYSY3LZ_A_u4woypvbihx5sEiaYvRSbdKKxIs0-ZKXE6CryQtFaJZGqXF7C9iA7tyyX3P4Ueu2T1REbV0MJni_St2-XOmwARV-zDCvfLSvet9jfMKaH2Sfb0tlle3lbqwMfLc73qIvswsWwJGdzl42c8weBzuh2z-1g935WNfT_iOhDmpR3iO96KIkE1S2DjoT-5Vgr0TxfSqXQlP6EjMQVRJ1oeYGfbRpipPEMvNncaqZLCA6z4pOmo8ZE4JN57OKXGEGd8ASx4S3fHNyVfvXTfHBeOe7jp3oFfF6wnEGYngsDIFUnvA",
    refreshToken: "1//04LkubD0qny1SCgYIARAAGAQSNwF-L9IryGTjtm1F7Y9m1NWLQLeWjwClB2yH70wy7lUBqYVhOyNPPSqpNp3CIbY2Z5K1bCymKRs",
    type: "success",
    user: {
        email: "laceyfamily203@gmail.com",
        familyName: "Family",
        givenName: "Lacey",
        id: 118123904703356740554,
        name: "Lacey Family",
        photoUrl: "https://lh4.googleusercontent.com/-zBCok_7BNp0/AAAAAAAAAAI/AAAAAAAAAAA/AAKWJJP1QwxK7WLx0hQ-vYcIHwY9u_pmRQ/photo.jpg",
    },
}
//have one task that's scheduled
//have one task that's just a potential </pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Fri Apr 17 2020 09:39:40 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    