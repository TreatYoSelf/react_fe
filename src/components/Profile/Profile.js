import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, Image, View, ScrollView, FlatList, TouchableOpacity } from 'react-native';
import SuggestedTreat from '../../containers/suggestedTreat/SuggestedTreat';
import { gql } from 'apollo-boost';
import { useQuery } from '@apollo/react-hooks';

export default function Profile(props) {
    //need to fetch all activities for a user, and map over them adding suggesteTreats for each
    //temporarily load SuggestedTreat
    let catDisplay, selectedCategories = [];
    const myQuery = gql`
        {
          getCategories {
            name
          }
        }
      `;

    const { error, data } = useQuery(myQuery, { errorPolicy: 'all' });
    //need to error display
    // if (error) {
    //     setErrors(error)
    // }

    //this needs to be refactored to a toggle.
    //store as object for easy removal 
    const selectCategory = (category) => {
        if (selectedCategories.length < 3) {
            selectedCategories.push(category)
        }
    }

    const submitSelection = () => {
        if (selectedCategories.length === 3) {
            console.log(selectedCategories)
        }
    }

    //render fetched categories
    if (data) {
        catDisplay = data.getCategories.map((category, i) =>
            <SuggestedTreat key={i} title={category.name} selectCategory={selectCategory} />)
    }

    return (
        <View style={styles.container}>
            <Text style={styles.header}>Welcome:{props.name}</Text>
            <Image style={styles.image} source={{ uri: props.photoUrl }} />
            <ScrollView style={styles.categories}>
                {catDisplay}
            </ScrollView>
            {/* <FlatList
                data={data.getCategories}
                ref={e => (this.items = e)}
                renderItem={({ category, i }) => (
                    <SuggestedTreat key={i} title={category.name} selectCategory={selectCategory} />
                )}
            /> */}
            <TouchableOpacity style={styles.container} onPress={() => submitSelection()}>
                <Text style={styles.header}>Submit</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#fff",
        alignItems: "center",
        // justifyContent: "space-between",
        // height: 40,
        // width: 100
    },
    header: {
        fontSize: 25,
    },
    image: {
        marginTop: 15,
        width: 150,
        height: 150,
        borderColor: "rgba(0,0,0,0.2)",
        borderWidth: 3,
        borderRadius: 150
    },
    categories: {
        backgroundColor: "#fff",
        flexGrow: 0,
        // height: 60
        // width: 100
    }
})
