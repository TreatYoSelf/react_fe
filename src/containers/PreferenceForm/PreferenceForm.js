import React, { useState } from 'react';
import { StyleSheet, Text, View, FlatList, TouchableOpacity } from 'react-native';
import TreatSelector from '../../containers/TreatSelector/TreatSelector';
import { gql } from 'apollo-boost';
import { useQuery } from '@apollo/react-hooks';
import { Link } from "react-router-native";

export const FETCH_CATEGORIES = gql`
    {
      getCategories {
        name
      }
    }
  `;

export default function PreferenceForm() {
    let catDisplay, allCategories = [];
    let selectedCategories = {};
    const [selectedStatus, setSelectionStatus] = useState(false);

    const { error, data } = useQuery(FETCH_CATEGORIES, { errorPolicy: 'all' });
    //need to error display
    // if (error) {
    //     setErrors(error)
    // }

    //this needs to be refactored to a toggle.
    //store as object for easy removal 

    //create toggle as two separate tasks. 
    //one should go through and as long as 

    //fix up naming convention here 
    const postPreferences = (chosenCat) => {
        const options = {
            headers: {
                'Content-Type': 'application/json'
            },
            method: "POST",
            body: JSON.stringify(chosenCat)
        }

        // fetch('https://treat-yo-self-bjtw.herokuapp.com/api/v1/suggestions', options)
        //     .then(resp => resp.json())
        //     .then(data => console.log('preferences posted'))
        //     .catch(err => console.log(err))
    }

    const selectCategory = (category, id) => {
        if (allCategories[id].isSelect) {
            allCategories[id].isSelect = false;
            delete selectedCategories[id];
        } else if (Object.keys(selectedCategories).length < 3) {
            selectedCategories[id] = category;
            allCategories[id].isSelect = true;
            if (Object.keys(selectedCategories).length === 3) {
                setSelectionStatus(true)
            }
        }
    }

    const submitSelection = () => {
            console.log('hit')
            // postPreferences(Object.keys(selectedCategories))
            //navigate back to profile also 
        //if failed submit error message
    }

    if (data) {
        allCategories = data.getCategories.map((category, index) => {
            return {
                id: index.toString(),
                isSelect: false,
                ...category
            }
        })

        catDisplay = <FlatList
            data={allCategories}
            renderItem={({ item }) => (
                <TreatSelector key={item.id} id={item.id} title={item.name} style={item.isSelect ? 'selected' : ''} selectCategory={selectCategory} />
            )}
        />
    }

    return (
        <View style={styles.container}>
            <Text style={styles.header}>What types of activities are you interested in?</Text>
            {catDisplay}
            {selectedStatus && <TouchableOpacity style={styles.button} onPress={() => submitSelection()} disabled={selectedStatus ? false : true}>
                <Text>Submit Choices</Text>
            </TouchableOpacity>}
            <Link to="/profile" style={styles.button}>
                <Text>Profile</Text>
            </Link>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        alignItems: "center",
        // justifyContent: "space-between",
        // height: 40,
        // width: 100
    },
    header: {
        fontSize: 25,
        color: '#003045'
    },
    image: {
        marginTop: 15,
        width: 150,
        height: 150,
        borderColor: "rgba(0,0,0,0.2)",
        borderWidth: 3,
        borderRadius: 150
    },
    categories: {
        backgroundColor: "#fff",
        flexGrow: 0,
        // height: 60
        // width: 100
    },
    button: {
        alignItems: 'center',
        justifyContent: 'space-around',
        flexDirection: 'row',
        width: 300,
        padding: 10,
        backgroundColor: "honeydew",
        borderWidth: 5,
    }
})
