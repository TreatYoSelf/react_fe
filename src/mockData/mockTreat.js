export const mockTreat = {
    id: 1,
    icon: 'https://www.pikpng.com/pngl/b/65-658548_source-munns-com-au-report-sun-icon-png.png',
    title: 'Frolic',
    rating: 3,
}

export const mockTreatSched = {
    id: 1,
    icon: 'https://www.pikpng.com/pngl/b/65-658548_source-munns-com-au-report-sun-icon-png.png',
    title: 'Competitive Plumbing',
    date: '',
    duration: ''
}

export const mockEvents = [{
        "eventName": "Hike",
            "eventStartTime": "1394648309130.185",
            "eventEndTime": "1394651309130.185"
    },
    {
        "eventName": "Spa Day",
            "eventStartTime": "1396678309130.185",
            "eventEndTime": "1396682309130.185"
    },
    {
        "eventName": "Yoga",
            "eventStartTime": "1396678509130.185",
            "eventEndTime": "1396682309130.185"
    }];

//have one task that's scheduled
//have one task that's just a potential 