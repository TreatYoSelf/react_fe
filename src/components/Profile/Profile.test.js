import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';

import Profile from './Profile';
import { FETCH_CATEGORIES } from '../../containers/PreferenceForm/PreferenceForm';
import { mockCategories } from '../../mockData/mockTreat';

describe('<Profile />', () => {
    let tree;
    const mocks = [
        {
            request: {
                query: FETCH_CATEGORIES,
            },
            result: {
                data: mockCategories
            },
        },
    ];

    beforeEach(async () => {
        await act(async () => {
        tree = await renderer.create(
            <MockedProvider mocks={mocks} addTypename={false}>
                 <Profile />
            </MockedProvider>
        ).toJSON();
        })
    })

    it('has 3 children', () => {
        expect(tree.children.length).toBe(3);
    });

    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });
});