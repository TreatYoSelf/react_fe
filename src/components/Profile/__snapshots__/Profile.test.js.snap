// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
      }
    }
  >
    Welcome:
  </Text>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderColor": "rgba(0,0,0,0.2)",
        "borderRadius": 150,
        "borderWidth": 3,
        "height": 150,
        "marginTop": 15,
        "width": 150,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
    <Text>
      Frolic
    </Text>
    <View
      style={
        Object {
          "height": 50,
          "width": 150,
        }
      }
    >
      <RCTPicker
        items={Array []}
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
